### PR TITLE
Update test docs after refactoring the instance manager lifecycle

### PR DESCRIPTION
**Pull request by shuo-wu**
https://github.com/longhorn/longhorn/issues/2650
https://github.com/longhorn/longhorn/issues/3459

### CODE REVIEW

**Review by innobead — APPROVED** [review]
LGTM

**Review by innobead — APPROVED** [review]
LGTM. Leave @longhorn/qa review then merge. 

@khushboo-rancher just a side note, probably we also need to adjust the drain command used in the drain test cases as per https://longhorn.io/docs/1.2.3/volumes-and-nodes/maintenance/#updating-the-node-os-or-container-runtime to include `--pod-selector='app!=csi-attacher,app!=csi-provisioner'`.

File: docs/content/manual/pre-release/node/kubelet-restart-on-a-node.md [hunk 1]
```diff
@@ -11,34 +11,36 @@ https://github.com/longhorn/longhorn/issues/2629
 3. Deploy prometheus monitoring app on the cluster which is using Longhorn storage class or deploy a statefulSet with Longhorn volume.
 4. Write some data into the mount point and compute the md5sum.
 5. Restart the kubelet on the node where the statefulSet or Prometheus pod is running using the command `sudo docker restart kubelet`
-6. Observe the volume. It gets detached and the pod gets terminated.
-7. Once the pod is terminated, a new pod should be created and get attached to the volume successfully.
-8. Verify that the mount of the volume is successful and data is safe.
+6. Observe the volume. It becomes degraded but is still running.
```
Comment (khushboo-rancher) [inline docs/content/manual/pre-release/node/kubelet-restart-on-a-node.md:14]: Just wanted to understand here, after restarting the kubelet on the node where the pod/volume is attached, will the engine not restart and result in volume detaching and attaching again?

File: docs/content/manual/pre-release/node/kubelet-restart-on-a-node.md [hunk 2]
```diff
@@ -11,34 +11,36 @@ https://github.com/longhorn/longhorn/issues/2629
...
 ### Case 2: Kubelet restart on one node RKE1 cluster:
 1. Create a RKE1 cluster with config of 1 node all role.
 2. Deploy Longhorn on the cluster.
 3. Deploy prometheus monitoring app on the cluster which is using Longhorn storage class or deploy a statefulSet with Longhorn volume.
 4. Write some data into the mount point and compute the md5sum.
 5. Restart the kubelet on the node using the command `sudo docker restart kubelet`
-6. Observe the volume. It gets detached and the pod gets terminated.
-7. Once the pod is terminated, a new pod should be created and get attached to the volume successfully.
-8. Verify that the mount of the volume is successful and data is safe.
+6. Check the instance manager pods on the node are still running.
```
Comment (khushboo-rancher) [inline docs/content/manual/pre-release/node/kubelet-restart-on-a-node.md:24]: Same as above doubt.

File: docs/content/manual/pre-release/node/kubelet-restart-on-a-node.md (hunk 1 above)
Comment (shuo-wu) [reply docs/content/manual/pre-release/node/kubelet-restart-on-a-node.md:14]: No. The volume won't become detached/crashed. Instead, it will keep running.

Comment (khushboo-rancher) [reply docs/content/manual/pre-release/node/kubelet-restart-on-a-node.md:14]: Great!

**Review by khushboo-rancher — APPROVED** [review]
LGTM

**Pull request merged**